pkg/controller/volume/expand/expand_controller.go

```diff
@@ -279,11 +279,7 @@ func (expc *expandController) syncHandler(key string) error {
 	if err != nil || volumePlugin == nil {
 		msg := fmt.Errorf("didn't find a plugin capable of expanding the volume; " +
 			"waiting for an external controller to process this PVC")
-		eventType := v1.EventTypeNormal
-		if err != nil {
-			eventType = v1.EventTypeWarning
-		}
-		expc.recorder.Event(pvc, eventType, events.ExternalExpanding, fmt.Sprintf("Ignoring the PVC: %v.", msg))
+		expc.recorder.Event(pvc, v1.EventTypeNormal, events.ExternalExpanding, fmt.Sprintf("Ignoring the PVC: %v.", msg))
```

Comment:
/assign @gnufied 

Should CSI return `err = nil`? What other scenarios is err returned?

Reply:
err is only returned when `FindPluginBySpec ` returns an error. 
https://github.com/kubernetes/kubernetes/blob/da9fdc933b3ac33a195bb5bf336e52bed2f790c5/pkg/volume/plugins.go#L652:28

An error is returned from that function in the following cases:
* Could not find plugin 
* no volume plugin matched
* multiple volume plugins matched

this function calls `CanSupport` for each volume driver to determine the matches. For CSI, this must be returning false? I'm not sure why, but here's the CSI's CanSupport implementation:
https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/csi/csi_plugin.go#L303